sdk/storage/azure-storage-common/src/storage_retry_policy.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "azure/storage/common/internal/storage_retry_policy.hpp"
+
+namespace Azure { namespace Storage { namespace _internal {
+
+  bool StorageRetryPolicy::ShouldRetryOnResponse(
+      Core::Http::RawResponse const& response,
+      Core::Http::Policies::RetryOptions const& retryOptions,
+      int32_t attempt,
+      std::chrono::milliseconds& retryAfter,
+      double jitterFactor) const
+  {
+    bool ret = RetryPolicy::ShouldRetryOnResponse(
+        response, retryOptions, attempt, retryAfter, jitterFactor);
+    if (ret)
+    {
+      return true;
+    }
+
+    if (attempt > m_options.MaxRetries)
```

Comment:
nit: should this use the local `retryOptions` variable?

Reply:
fixed